src/lib/php/Dao/LicenseDao.php

```diff
@@ -713,4 +713,34 @@ public function getLicenseObligations($licenseLists, $candidate = false)
       return $ObligationRef;
     }
   }
+
+  /**
+   * @param string $licenseShortname
+   * @param int|null $groupId
+   * @return null
+   */
+  public function deleteLicenseByShortName($licenseShortname, $groupId=null)
+  {
+    $extraCondition = "";
+    $condition = 'rf_shortname=$1';
+    $param = array($licenseShortname);
+    $this->dbManager->deleteSingleRow(
+      "DELETE FROM license_ref WHERE $condition",
+      $param, __METHOD__ . ".$condition.only");
+    if (isset($groupId)) {
+      $userId = (isset($_SESSION) && array_key_exists('UserId', $_SESSION)) ? $_SESSION['UserId'] : 0;
+      if (!empty($userId)) {
+        $param[] = $userId;
+        $extraCondition = "AND group_fk IN (SELECT group_fk FROM group_user_member WHERE user_fk=$".count($param).")";
+      }
+      if (is_int($groupId) && empty($userId)) {
+        $param[] = $groupId;
+        $extraCondition = "AND group_fk=$".count($param);
+      }
```

Comment:
Sorry but I am not sure what's happening here.
1. You are reading user id from session? It might fail for REST API calls, consider reading it as a parameter.
2. If user id exists, delete the license for all groups it might exist for me. Otherwise delete it just for one group.
  - Can you please explain the rational behind it?

Reply:
@GMishx , thanks for your feedback. I am trying to make changes as per your suggestions.

Reply:
Hey @its-sushant , I am still not clear about these operations. If you have the group id for the user, why not just delete the license for that particular group instead of deleting it for all groups?

In the OpenAPI doc, mention it explicitly that the license will only be deleted for active group.